sql/2022/http/sites_using_link_preload_header.sql

```diff
@@ -0,0 +1,48 @@
+#standardSQL
+
+# Number of <link rel="preload">.
+
+CREATE TEMPORARY FUNCTION getNumLinkRelPreloadRespHeader(HTTPheaders STRING, header STRING)
+RETURNS FLOAT64 LANGUAGE js AS """
+try {
+  var headers = JSON.parse(HTTPheaders);
+
+  // Filter by header name (which is case insensitive)
+  // If multiple headers it's the same as comma separated
+  const allLinkHeaderValues = headers.filter(h => h.name.toLowerCase() == header.toLowerCase()).map(h => h.value);
+  let numPreload = 0;
+  for (let linkHeader of allLinkHeaderValues) {
+    const linkHeaderTokens = linkHeader.split(',');
+    for (let linkHeaderToken of linkHeaderTokens) {
+      const linkTokenProps = linkHeaderToken.split(';');
+      for (let linkTokenProp of linkTokenProps) {
+        if (linkTokenProps.includes('rel=preload')) {
+          numPreload++;
+        }
+      }
+    }
+  }
+  return numPreload;
+} catch (e) {
+  return [];
```

Comment:
The return value is numeric so this type should be comparable
```suggestion
  return 0;
```

Reply:
done

Reply:
I don't think we want -1 since an error would undo 1 valid count